File: src/p226_invert_binary_tree.rb
# frozen_string_literal: true

#
# @lc app=leetcode id=226 lang=ruby
#
# [226] Invert Binary Tree
#

require_relative 'structs.rb'

# @lc code=start
# @param {TreeNode} root
# @return {TreeNode}
def invert_tree(root)
  unless root.nil?
    root.left, root.right = invert_tree(root.right), invert_tree(root.left)
  end

  root
end
# @lc code=end

File: src/p226-invert-binary-tree.js
/*
 * @lc app=leetcode id=226 lang=javascript
 *
 * [226] Invert Binary Tree
 */

import { TreeNode } from "./structs.js";

// @lc code=start
/**
 * @param {TreeNode | null} root
 * @return {TreeNode | null}
 */
const invertTree = (root) => {
    if (root !== null) {
        const { left, right } = root;
        root.left = invertTree(right);
        root.right = invertTree(left);
    }

    return root;
};
// @lc code=end
